packages/fgtclb/academic-persons/Classes/Controller/ProfileController.php

```diff
@@ -180,6 +181,18 @@ public function detailAction(?Profile $profile = null): ResponseInterface
         $event = $this->eventDispatcher->dispatch(new ModifyDetailProfileEvent($profile, $this->view));
         $profile = $event->getProfile();
 
+        // Add page title based on profile name
+        $titleParts = array_filter([
```

Comment:
I'd say we should have a first round through the array elements and trimming leading & trailing spaces first to avoid "multiple" spaces. And it would ensure that only spaces are reduced to an empty string `""` which will be filtered out by the array_filter. In the current form, for example a FirstName with an empty space would not be filtered out and leading to two spaces when imploded afterwards.

Reply:
Moved this logic into the ProfileTitleProvider class now, reworked logic and covered with tests to tackle mentioned points.